module/VuFind/src/VuFind/Db/Service/UserServiceInterface.php

```diff
@@ -47,7 +47,17 @@ interface UserServiceInterface extends DbServiceInterface
      *
      * @param string $id ID.
      *
-     * @return UserEntityInterface
+     * @return ?UserEntityInterface
      */
     public function getUserById($id);
+
+    /**
+     * Retrieve a user object from the database based on the given field.
+     *
+     * @param string $fieldName  Field name
+     * @param string $fieldValue Field value
+     *
+     * @return ?UserEntityInterface
+     */
+    public function getUserByField($fieldName, $fieldValue);
```

Comment:
Is it worth adding typehints to the method definition while we're at it? Might be good to do the same for getUserById, though that's perhaps out of scope for this PR.

Reply:
@ThoWagen, I think it's generally desirable to add typehints to new code when possible. We haven't introduced them globally due to backward-compatibility concerns, but someday it would be nice to increase the level of type safety in the code, and if we add hints as we refactor, it will make that eventual job easier when we get there. :-)